apps/provisioning_api/lib/Users.php

```diff
@@ -628,22 +628,39 @@ public function removeSubAdmin($parameters) {
 	 * @return Result
 	 */
 	public function getUserSubAdminGroups($parameters) {
-		$user = $this->userManager->get($parameters['userid']);
+
+		// Check if user is logged in
+		$currentLoggedInUser = $this->userSession->getUser();
+		if ($currentLoggedInUser === null) {
+			return new Result(null, API::RESPOND_UNAUTHORISED);
+		}
+
+		$currentLoggedInUserId = $currentLoggedInUser->getUID();
+
+		$targetUserId = $parameters['userid'];
+		$targetUser = $this->userManager->get($targetUserId);
+
 		// Check if the user exists
-		if ($user === null) {
+		if ($targetUser === null) {
 			return new Result(null, 101, 'User does not exist');
```

Comment:
```suggestion
			return new Result(null, API::RESPOND_NOT_FOUND, 'The requested user could not be found');
```

For consistency purposes